### ISSUE TITLE
[CBSLocalArticle] Unable to extract Anvato player data

### ISSUE THREAD

**Issue by sogilbert** [yt-dlp/yt-dlp#2794]
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.02.04**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

So I tried to download this news story, but it appears to be complaining that it can't extract Avano player data. Just as a side note, I checked the embed link, and it goes to a Luna.live domain. Would really appreciate if you checked it out.

### Verbose log

```shell
[debug] Command-line config: ['--verbose', 'https://cbsloc.al/3LkEwK8']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.02.04 [c1653e9ef]
[debug] Python version 3.10.1 (CPython 32bit) - Linux-4.4.177-22626479-armv8l-with-libc
[debug] exe versions: ffmpeg 4.4.1 (setts), ffprobe 4.4.1
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://cbsloc.al/3LkEwK8
[generic] 3LkEwK8: Requesting header
[redirect] Following redirect to https://miami.cbslocal.com/2022/02/12/no-appetite-for-new-miami-restaurant-glorifying-castro-communism/
[debug] [CBSLocalArticle] Extracting URL: https://miami.cbslocal.com/2022/02/12/no-appetite-for-new-miami-restaurant-glorifying-castro-communism/
[CBSLocalArticle] no-appetite-for-new-miami-restaurant-glorifying-castro-communism: Downloading webpage
ERROR: [CBSLocalArticle] no-appetite-for-new-miami-restaurant-glorifying-castro-communism: Unable to extract Anvato player data; please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the "Broken site" issue template properly. Confirm you are on the latest version using -U; please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the "Broken site" issue template properly. Confirm you are on the latest version using -U
  File "/data/data/com.termux/files/usr/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 612, in extract
    ie_result = self._real_extract(url)
  File "/data/data/com.termux/files/usr/lib/python3.10/site-packages/yt_dlp/extractor/cbslocal.py", line 107, in _real_extract
    info_dict = self._extract_anvato_videos(webpage, display_id)
  File "/data/data/com.termux/files/usr/lib/python3.10/site-packages/yt_dlp/extractor/anvato.py", line 382, in _extract_anvato_videos
    self._html_search_regex(
  File "/data/data/com.termux/files/usr/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1195, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "/data/data/com.termux/files/usr/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1186, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
```

**Comment by dirkf** [comment]
Anvato was bought up by Google. The legacy streaming support now seems to be at `mp.lura.live`. So all this needs a rework.

First, though, the page itself has a structure not supported by the `CBSLocalArticleIE` extractor, so the Anvato access key and video ID have to be extracted in a different way. There is a script inside the `section#featured-media` element containing the required fields as the `accessKey` and `video` members of JSON that is passed to `AnvatoPlayer('p0').init()`. The page also has a nice `ld+json` script element that can be extracted for metadata.

Additionally the extractor first tries to extract a URL suitable for the `SendtoNewsIE` extractor. This doesn't succeed with the problem URL (or any other current CBSLocal page that I could find) but that extractor also needs a rework.

As for the `AnvatoIE` extractor, a similar pattern of requests to the original is needed, but with slightly different parameters and different API URLs.
1. From `'https://access.mp.lura.live/anvacks/%s?apikey=%s' % (access_key, self._API_KEY)`, get JSON containing all the API URLs with replacement fields that have to be replaced with the video parameters.
2. Use the `api.time` member to get the server time (UNIX epoch).
3. Use the `api.video` member to get video metadata JSON using query parameters and POST data calculated, similar to the existing, from the video data URL and server time.

Fortunately this has the same format as the JSON expected from the old Anvato API, so we just carry on.

Obviously this is too complicated to be offered as a patch, so PR(s) will be needed.

**Comment by bashonly** [comment]
OP's URL is dead, but everything that could be fixed with `CBSLocal`/`CBSLocalArticle` has been resolved by f6e43d6fa9804c24525e1fed0a87782754dab7ed
